### ISSUE TITLE
Erroneous not-an-iterable when inheriting from typing generics

### ISSUE THREAD

**Issue by keis** [PyCQA/pylint#2568]
### Steps to reproduce
1. Lint the following
```py
from typing import Tuple, List


class Foo(Tuple[str, ...]):
    pass


class Bar(List[str]):
    pass


for x in Foo('ab'):
    pass

for x in Bar('ab'):
    pass
```

### Current behavior
Reported as not iterable
```
foo.py:12:9: E1133: Non-iterable value Foo('ab') is used in an iterating context (not-an-iterable)
foo.py:15:9: E1133: Non-iterable value Bar('ab') is used in an iterating context (not-an-iterable)
```

### Expected behavior
Similar to inheriting from plain `tuple` or `list` no error should be reported

### pylint --version output
```
pylint 2.1.1
astroid 2.1.0-dev
Python 3.6.6 (default, Jul 19 2018, 14:25:17) 
[GCC 8.1.1 20180712 (Red Hat 8.1.1-5)]
```

**Comment by PCManticore** [comment]
Thanks for creating an issue!

**Comment by jstriebel** [comment]
Similar minimal example with `NamedTuple` (tested with python 3.7.2):
```
from __future__ import annotations
from typing import NamedTuple

class Vec3D(NamedTuple):
    x: int
    y: int
    z: int

    def inc(self) -> Vec3D:
        return Vec3D(a + 1 for a in self)
```
produces `E1133: Non-iterable value self is used in an iterating context (not-an-iterable)`.

**Comment by Greyzerus** [comment]
Same when using self

```
import typing

class Foo(typing.List[str]):
    def foo():
         for bar in self:
            print(bar)
```
`[Pylint: E1133(not-an-iterable), Foo.foo] Non-iterable value self is used in an iterating context`

**Comment by jacobtylerwalls** [comment]
Can't reproduce on main @ 314069664138ed00be9a08af642135ffe0dbbf9f
